tests/unittest.py

```diff
@@ -223,21 +215,92 @@ def assertIsTopological(self, tx_sequence: Iterator[BaseTransaction], message: O
+                                                      'the test class or pass `enable_sync_v2` by argument')
+            enable_sync_v2 = self._enable_sync_v2
+        assert enable_sync_v1 or enable_sync_v2, 'enable at least one sync version'
+        return enable_sync_v1, enable_sync_v2
+
     def assertTipsEqual(self, manager1, manager2):
+        _, enable_sync_v2 = self._syncVersionFlags()
+        if enable_sync_v2:
+            self.assertTipsEqualSyncV2(manager1, manager2)
+        else:
+            self.assertTipsEqualSyncV1(manager1, manager2)
+
+    def assertTipsNotEqual(self, manager1, manager2):
         s1 = set(manager1.tx_storage.get_all_tips())
         s2 = set(manager2.tx_storage.get_all_tips())
-        self.assertEqual(s1, s2)
+        self.assertNotEqual(s1, s2)
 
-        s1 = set(manager1.tx_storage.get_tx_tips())
-        s2 = set(manager2.tx_storage.get_tx_tips())
+    def assertTipsEqualSyncV1(self, manager1, manager2):
+        # tx tips
+        tips1 = {tx.hash for tx in manager1.tx_storage.iter_mempool_tips_from_tx_tips()}
+        tips2 = {tx.hash for tx in manager2.tx_storage.iter_mempool_tips_from_tx_tips()}
+        self.log.debug('tx tips1', len=len(tips1), list=shorten_hash(tips1))
+        self.log.debug('tx tips2', len=len(tips2), list=shorten_hash(tips2))
+        self.assertEqual(tips1, tips2)
+
+        # best block
+        s1 = set(manager1.tx_storage.get_best_block_tips())
+        s2 = set(manager2.tx_storage.get_best_block_tips())
         self.assertEqual(s1, s2)
 
-    def assertTipsNotEqual(self, manager1, manager2):
+        # best block (from height index)
+        b1 = manager1.tx_storage.indexes.height.get_tip()
+        b2 = manager2.tx_storage.indexes.height.get_tip()
+        self.assertEqual(b1, b2)
+
+        # all tips must be equal (this check should be removed together with the index)
         s1 = set(manager1.tx_storage.get_all_tips())
         s2 = set(manager2.tx_storage.get_all_tips())
-        self.assertNotEqual(s1, s2)
+        self.assertEqual(s1, s2)
+
+    def assertTipsEqualSyncV2(self, manager1, manager2, *, strict_sync_v2_indexes=True):
+        # tx tips
+        if strict_sync_v2_indexes:
+            tips1 = manager1.tx_storage.indexes.mempool_tips.get()
+            tips2 = manager2.tx_storage.indexes.mempool_tips.get()
+        else:
+            tips1 = {tx.hash for tx in manager1.tx_storage.iter_mempool_tips_from_best_index()}
+            tips2 = {tx.hash for tx in manager2.tx_storage.iter_mempool_tips_from_best_index()}
+        self.log.debug('tx tips1', len=len(tips1), list=shorten_hash(tips1))
+        self.log.debug('tx tips2', len=len(tips2), list=shorten_hash(tips2))
+        # self.assertEqual(tips1, tips2)
+        # XXX: this is still not correct, tips may diverge since voided transactions aren't propagated
+        # XXX: temporarily disabled because tips may not match since sync-v2-mempool isn't merged yet
```

Comment:
Outdated comment? Should we uncomment `self.assertEqual(tips1, tips2)`?

Reply:
Done.

Reply:
Thanks!